chainer/links/activation/dropconnect.py

```diff
@@ -0,0 +1,86 @@
+import math
+
+from chainer import cuda
+from chainer.functions.noise import dropconnect
+from chainer import initializers
+from chainer import link
+
+
+class Dropconnect(link.Link):
+
+    """Fully-connected layer with dropconnect regularization.
+
+    Args:
+        in_size (int): Dimension of input vectors. If None, parameter
+            initialization will be deferred until the first forward data pass
+            at which time the size will be determined.
+        out_size (int): Dimension of output vectors.
+        wscale (float): Scaling factor of the weight matrix.
+        initialW (3-D array or None): Initial weight value.
+            If ``None``, then this function uses ``wscale`` to initialize.
+        initial_bias (2-D array, float or None): Initial bias value.
+            If it is float, initial bias is filled with this value.
+            If it is ``None``, bias is omitted.
+        mask (:class:`numpy.ndarray` or cupy.ndarray):
+            If ``None``, randomized dropconnect mask is generated.
+            If not ``None``, this value is used as dropconnect mask.
+            And scaling will not be executed.
+            Then mask shape must be ``(M, N)``.
+            Main purpose of latter option is debugging.
+
+    Attributes:
+        W (~chainer.Variable): Weight parameter.
+        b (~chainer.Variable): Bias parameter.
+
+    .. seealso:: :func:`~chainer.functions.dropconnect`
+
+    .. seealso::
+        Li, W., Matthew Z., Sixin Z., Yann L., Rob F. (2013).
+        Regularization of Neural Network using DropConnect.
+        International Conference on Machine Learning.
+        `URL <http://cs.nyu.edu/~wanli/dropc/>`_
+    """
+
+    def __init__(self, in_size, out_size, wscale=1,
+                 ratio=.5, initialW=None, initial_bias=0, mask=None):
```

Comment:
Initially, I implemented `mask` option to `__call__`. So the alternative is natural for me.

But I found it is difficult that setting `ndarray` type `mask` by `gradient_check.check_backward`.
Because `check_backward` convert `ndarray` to `Variable`.
And current version `functions.dropconnect` expects `mask` to be `ndarray`.

I know it is not problem if I change `dropconnect` to allow `Variable` type `mask`.
Add to `__call__` not `__init__` is better?